### PR TITLE
Remove diff and request data

### PR DESCRIPTION
**Pull request by chouseknecht**
##### SUMMARY
- Remove request and diff data from module output in an attempt to fully address security concerns raised by [openshift-restclient-python #107](https://github.com/openshift/openshift-restclient-python/issues/107)
- Add more RETURN docs detail 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/k8s/common.py
lib/ansible/modules/clustering/k8s/k8s_raw.py 
lib/ansible/modules/clustering/openshift/openshift_raw.py

##### ANSIBLE VERSION
```
ansible 2.5.0 (devel 2d13167f1c) last updated 2017/12/25 21:34:22 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/Users/chouseknecht/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/chouseknecht/projects/ansible/lib/ansible
  executable location = /Users/chouseknecht/.pyenv/versions/venv27/bin/ansible
  python version = 2.7.14 (default, Nov 14 2017, 23:24:24) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.38)]
```

### CODE REVIEW

**Comment by towolf** [comment]
thumbs down

File: lib/ansible/module_utils/k8s/common.py [hunk 1]
```diff
@@ -167,9 +167,6 @@ def execute_module(self):
 
         return_attributes = dict(changed=False, result=dict())
 
-        if self._diff:
```
Comment (towolf) [inline lib/ansible/module_utils/k8s/common.py:170]: This is bad.

There is an if condition here, controlled by the user.

Why remove this option from the user?

**Comment by chouseknecht** [comment]
@towolf 

The intention of the `--diff` parameter is to show differences in files and templates. It wasn't really intended to show debugging type info as we were doing here.

But nevertheless, the primary reason for removing request and diff data, as mentioned in the description, is to make sure we're not accidentally logging sensitive data. I think given there was no attempt being made to filter out anything sensitive, removing access altogether is the right action.

**Comment by towolf** [comment]
Thanks for the info @chouseknecht.

I still beg to differ on the "security" aspect. My machine is trusted, and I do not have "shoulder surfers".

And secrets fly across my screen all day long.

I contributed about 10 PR's to Ansible to add `--diff` mode. Seeing someone remove it makes we whince ...

**Pull request merged**